### ISSUE TITLE
[Quick Add] Enable 1-click Quick Add for any product in product grid where only 1 option exists

### ISSUE THREAD

**Issue by bredowmax** [Shopify/dawn#1604]
**Describe the current behavior**
I'm looking into transitioning from the Warehouse theme to Dawn but came across one major limitation regarding Quick Add #1388: Similar to the Empire theme, Dawn appears to be incompatible with Product Options. In exercise, there is no "Add to Cart" button for products that have exactly 1 variant defined.

In the store I'm talking about, shoppers on average buy 10-20 products, and select their products mainly from the collection grid, since products are low-consideration, such as "Banana". 

Assuming 2 clicks difference between the "Add to Cart" vs. "Open Modal" versions of the Quick Add button, and 15 products per cart, a shopper saves 30 clicks with the "Add to Cart" version vs. the "Open Modal" version.

The Warehouse theme does _not_ have this limitation, and with a product inventory of 500 different SKUs, it would be quite an effort for the merchant to make the inventory compatible with the new Quick Add feature.

**Describe the expected behavior**
For any product with only 1 option, I'd expect shoppers to be able to add this variant to the cart with only one click. That means: _without_ opening the Quick Add modal.

I'd expect the same behavior for products where several variants exist, but only 1 option is valid with applied facet filters. Example: A shirt which exists in Blue, Red & Yellow, while color facet filter is set to "blue". An additional message would say: _"More colors available"_

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 4.0.0

**Possible solution**
Option 1: 
- My proposed solution treats any product with only 1 variant the same - no matter if this one variant was specifically defined or not. 

Option 2:
- I wouldn't be able to produce an example, but I could imagine that some merchants may want to have this particular setup (that is: opening a Quick Add modal even if only 1 variant is available). Therefore, option 2 proposes a theme setting to let merchants choose the preferred behavior

**Additional context/screenshots**
<img width="698" alt="Screen Shot 2022-04-09 at 4 09 45 PM" src="https://user-images.githubusercontent.com/30573532/162590120-784f945e-5fae-4344-a1fe-c7d7fff7a241.png">
<img width="1445" alt="Screen Shot 2022-04-09 at 4 09 24 PM" src="https://user-images.githubusercontent.com/30573532/162590121-b69859fe-aef4-4646-8177-43c9cae0be19.png">
<img width="341" alt="Screen Shot 2022-04-09 at 4 09 01 PM" src="https://user-images.githubusercontent.com/30573532/162590123-2391716f-f525-40e1-9b05-ce80c2bdd6ea.png">

**Comment by bredowmax** [comment]
... I initially thought I could transform my products so that the information I had used in Product Option 1 (quantity, such as _"piece, 1L, 6 pcs., XL, bottle"_) as a new defined variant Metafield - but now realized that I can't have a Variant Metafield unless I have a Product Option. 

This would limit me to Product Metafields. But I've already reserved Product Metafields for more general characteristics that are universally true for all possible variants, such as "allergens" or "SEO hidden" or "Eco label". 

In short - the workaround Variant Metafields doesn't get me further - I'm facing the same issues
